src/modules/store/modules/layers.store.js

```diff
@@ -39,10 +39,10 @@ const getters = {
      * @param state
      * @returns {AbstractLayer[]} All layers that are currently visible on the map
      */
-    visibleLayers: (state) =>
-        [...state.activeLayers.filter((layer) => layer.visible), state.previewLayer].filter(
-            (layer) => layer !== null
-        ),
+    visibleLayers: (state) => {
+        const activeVisible = state.activeLayers.filter((layer) => layer.visible)
+        return state.previewLayer === null ? activeVisible : [...activeVisible, state.previewLayer]
```

Comment:
instead of a ternary operator, you could use a `if` block that pushes the preview layer if defined
```
const activeVisible = state.activeLayers.filter((layer) => layer.visible)
if (state.previewLayer) {
   activeVisible.push(state.previewLayer)
}
return state.previewLayer
```
I find that more readable than a ternary (even though it achieves the same result)

Reply:
Works for me :+1: I'll change it.